state/geth_test.go

```diff
@@ -106,3 +107,25 @@ func TestGethDbReloadData(t *testing.T) {
 		t.Fatalf("Failed to close DB: %v", err)
 	}
 }
+
+func TestGethDb_CreateAccountIsProtected(t *testing.T) {
+	dir := t.TempDir()
+	db, err := MakeGethStateDB(dir, "", common.Hash{}, false, nil)
+	require.NoError(t, err)
+	addr := common.Address{0x22}
+	// First create the account
+	db.CreateAccount(addr)
+	// Then recall it - it must not panic
+	db.CreateAccount(addr)
+}
+
+func TestGethDb_CreateContractIsProtected(t *testing.T) {
+	dir := t.TempDir()
+	db, err := MakeGethStateDB(dir, "", common.Hash{}, false, nil)
+	require.NoError(t, err)
+	addr := common.Address{0x22}
+	// First create the contract
+	db.CreateContract(addr)
+	// Then recall it - it must not panic
+	db.CreateContract(addr)
+}
```

Comment:
Create a second test case for CreateContract() that calls CreateContract() without a prior CreateAccount() invocation. Assert that the address does not exist before and after the call to CreateContract(). This should be a negative test for ContractCreate() which should make geth crash if the guards are not enabled in the state interface.